DataDog/datadog-agent
# Add SNMP corecheck integration

### What does this PR do?

Add snmp corecheck integration

The core check implement has the same feature as Python snmp integration (except for MIB/autodiscovery support and few lower level configurations that have been excluded).

### High level design
![image](https://user-images.githubusercontent.com/49917914/105340394-1bab2380-5bde-11eb-9966-73eb991900ba.png)

### Motivation

Python snmp integration has performance issue, snmp core check is faster and have lower memory/cpu usage.

### Describe your test plan

1/ unit tests

2/ testing with python snmp e2e test suite to verify feature parity with python implementation

See https://github.com/DataDog/integrations-core/pull/8048


### Manual Testing

Using image `datadog/agent-dev:alex-snmp-corecheck`

```
ddev env start snmp py38-python_ad --agent datadog/agent-dev:alex-snmp-corecheck

ddev env shell snmp py38-python_ad -v
```

Change `/etc/datadog-agent/conf.d/snmp.d/snmp.yaml` of the agent container to for example:

```yaml
init_config:
instances:
  - loader: core
    ip_address: XXXX
    port: 1161
    community_string: f5-big-ip
```

Then run

```
$ agent check snmp
```
